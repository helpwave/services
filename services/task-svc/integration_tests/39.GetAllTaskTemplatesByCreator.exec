#!/bin/bash

grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"created_by": "df363387-7774-4592-b01c-2b765abaa86a"}' \
	localhost:3001 \
	services.task_svc.v1.TaskTemplateService/GetAllTaskTemplatesByCreator

grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"created_by": "5db6cc86-bbf1-4fa6-bbe8-a8cdea869758", "private_only": true}' \
	localhost:3001 \
	services.task_svc.v1.TaskTemplateService/GetAllTaskTemplatesByCreator
