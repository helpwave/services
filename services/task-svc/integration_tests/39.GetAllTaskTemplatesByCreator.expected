{
  "templates": [
    {
      "id": "ff2e6ec7-f8d1-4c39-b5e5-6bb5ad99a1f1",
      "name": "Supplement Vagina with Nonaut Sub, Via Opening",
      "description": "Autologous hematopoietic stem cell transplant with purging",
      "isPublic": true,
      "subtasks": [
        {
          "id": "3845c063-17d7-4120-a5be-e91bec099f77",
          "taskTemplateId": "ff2e6ec7-f8d1-4c39-b5e5-6bb5ad99a1f1",
          "name": "Supplement Skull with Synthetic Substitute, Perc Approach"
        }
      ]
    },
    {
      "id": "12dedb6e-de87-471d-b812-44fbfaa9e73f",
      "description": "Insertion of therapeutic device into uterus",
      "isPublic": true,
      "subtasks": [
        {
          "id": "fe6b8a5e-bffc-4cc8-bdfb-883bd6392ac2",
          "taskTemplateId": "12dedb6e-de87-471d-b812-44fbfaa9e73f",
          "name": "Bypass R Ventricle to L Pulm Art w Nonaut Sub, Perc Endo"
        }
      ]
    },
    {
      "id": "dd3f7c98-2b61-4b61-922a-d730bfb53ae7",
      "name": "Bypass 1 Cor Art from R Int Mammary w Synth Sub, Perc Endo",
      "description": "Paraurethral suspension",
      "isPublic": true,
      "subtasks": [
        {
          "id": "2767aa0d-a4dc-4cda-8d4f-030b272b6ede",
          "taskTemplateId": "dd3f7c98-2b61-4b61-922a-d730bfb53ae7"
        }
      ]
    },
    {
      "id": "d9985e6e-2240-4f4f-8774-ed0b60141ac0",
      "name": "LDR Brachytherapy of Pancreas using Palladium 103",
      "isPublic": true,
      "subtasks": [
        {
          "id": "466d13cf-7c15-4474-bb36-1a47073fc216",
          "taskTemplateId": "d9985e6e-2240-4f4f-8774-ed0b60141ac0",
          "name": "Drain of Accessory Sinus with Drain Dev, Perc Endo Approach"
        }
      ]
    },
    {
      "id": "e82fb996-e70b-45e5-90d6-889ede6a0d0f",
      "name": "Dilation of Lingula Bronchus, Open Approach",
      "isPublic": true
    },
    {
      "id": "40297575-f54d-4d77-b6e2-e0b680996e51",
      "name": "Removal of Extralum Dev from Hepatobil Duct, Open Approach",
      "isPublic": true
    }
  ]
}
{
  "templates": [
    {
      "id": "ee863790-1faa-4610-b845-39ba878a59fe",
      "name": "Muscle Perform Trmt Neuro Up Back/UE w Prosthesis",
      "subtasks": [
        {
          "id": "0840c5e4-835e-493f-a15d-ef0e03aefb0b",
          "taskTemplateId": "ee863790-1faa-4610-b845-39ba878a59fe",
          "name": "Dilate R Com Iliac Art w 2 Intralum Dev, Perc Endo"
        },
        {
          "id": "0956f068-73d0-4035-a015-a38594b28ef4",
          "taskTemplateId": "ee863790-1faa-4610-b845-39ba878a59fe"
        }
      ]
    }
  ]
}
